lib/mix/lib/mix/gleam.ex

```diff
@@ -0,0 +1,118 @@
+defmodule Mix.Gleam do
+  # Version that introduced `gleam export package-information` command
+  @required_gleam_version ">= 1.10.0"
+
+  def load_config(dir) do
+    File.cd!(dir, fn ->
+      gleam!(["export", "package-information", "--out", "/dev/stdout"])
+      |> JSON.decode!()
+      |> Map.fetch!("gleam.toml")
+      |> parse_config()
+    end)
+  end
+
+  def parse_config(json) do
+    try do
+      deps =
+        Map.get(json, "dependencies", %{})
+        |> Enum.map(&parse_dep/1)
+
+      dev_deps =
+        Map.get(json, "dev-dependencies", %{})
+        |> Enum.map(&parse_dep(&1, only: :dev))
+
+      %{
+        name: Map.fetch!(json, "name"),
+        version: Map.fetch!(json, "version"),
+        deps: deps ++ dev_deps
+      }
+      |> maybe_gleam_version(json)
+      |> maybe_erlang_opts(json)
+    rescue
+      KeyError ->
+        Mix.raise("Command \"gleam export package-information\" unexpected format: \n" <> json)
+    end
+  end
```

Comment:
We can do away without `try`.

```suggestion
  def parse_config(json) do
    deps =
      Map.get(json, "dependencies", %{})
      |> Enum.map(&parse_dep/1)

    dev_deps =
      Map.get(json, "dev-dependencies", %{})
      |> Enum.map(&parse_dep(&1, only: :dev))

    %{
      name: Map.fetch!(json, "name"),
      version: Map.fetch!(json, "version"),
      deps: deps ++ dev_deps
    }
    |> maybe_gleam_version(json)
    |> maybe_erlang_opts(json)
  rescue
    KeyError ->
      Mix.raise("Command \"gleam export package-information\" unexpected format: \n" <> json)
  end
```